rtvi-client-js/src/client.ts

```diff
@@ -0,0 +1,1031 @@
+import cloneDeep from "clone-deep";
+import EventEmitter from "events";
+import TypedEmitter from "typed-emitter";
+
+import packageJson from "../package.json";
+import { getIfTransportInState, transportReady } from "./decorators";
+import * as RTVIErrors from "./errors";
+import { RTVIEvent, RTVIEvents } from "./events";
+import { RTVIClientHelper, RTVIClientHelpers } from "./helpers";
+import {
+  BotLLMTextData,
+  BotReadyData,
+  ConfigData,
+  MessageDispatcher,
+  PipecatMetricsData,
+  RTVIActionRequest,
+  RTVIActionRequestData,
+  RTVIActionResponse,
+  RTVIMessage,
+  RTVIMessageType,
+  StorageItemStoredData,
+  TranscriptData,
+} from "./messages";
+import { Participant, Tracks, Transport, TransportState } from "./transport";
+
+export type ConfigOption = {
+  name: string;
+  value: unknown;
+};
+
+export type RTVIClientConfigOption = {
+  service: string;
+  options: ConfigOption[];
+};
+
+export type RTVIURLEndpoints = "connect" | "action";
+
+const defaultEndpoints: Record<RTVIURLEndpoints, string> = {
+  connect: "/connect",
+  action: "/action",
+};
+
+export type RTVIClientParams = {
+  baseUrl: string;
+} & Partial<{
+  headers?: Headers;
+  endpoints: Record<RTVIURLEndpoints, string>;
+  requestData?: object;
+  config?: RTVIClientConfigOption[];
+}> & {
+    [key: string]: unknown;
```

Comment:
What are the extra optional fields for here? This pattern isn't possible on mobile.